src/profile/steps.js

```diff
@@ -62,22 +62,39 @@ const recovery = {
 
 const isAlternate = (method) => method.type === 'email'
 
-const twosv = {
-  nameKey: 'profile.steps.2svStep',
+const totp = {
+  nameKey: 'profile.steps.authenticator',
   paths: [
     '/2sv/intro',
     '/2sv/smartphone/intro',
     '/2sv/smartphone/download-app',
     '/2sv/smartphone/scan-qr',
     '/2sv/smartphone/verify-qr-code',
     '/2sv/smartphone/code-verified',
+  ],
+  isRelevant(user, recoveryMethods, mfa) {
+    return user.auth_type === 'login' && (isRequested(this.paths) || mfa.numVerified < 3)
+  },
+}
+
+const securityKeyStep = {
+  nameKey: 'profile.steps.securityKeyStep',
+  paths: [
     '/2sv/usb-security-key/intro',
     '/2sv/usb-security-key/insert',
     '/2sv/usb-security-key/touch',
     '/2sv/usb-security-key/confirmed',
-    '/2sv/printable-backup-codes/intro',
-    '/2sv/printable-backup-codes/new',
   ],
+  isRelevant(user, recoveryMethods, mfa) {
+    const numberOfKeys = (mfa.keys.data?.length || 0) + (mfa.u2f?.id ? 1 : 0)
+
+    return user.auth_type === 'login' && (isRequested(this.paths) || numberOfKeys === 0)
```

Comment:
Would it make sense to change this to `numberOfKeys < 2` so they get prompted when they're low on codes? It might require a change to the wording on that page. If so, don't worry about it.

Reply:
do you mean mfa.numVerified? keys are just u2f and webauthn

Reply:
My comment doesn't make sense here. I was thinking about backup codes, which would be on line 99. And to check the number of remaining codes is more complex and I'm not exactly sure how it would work.